src/ServerHandler.cpp

```diff
@@ -1,8 +1,153 @@
 //
-// Created by idanto on 11/01/2020.
+// Created by idanto on 13/01/2020.
 //
 
 #include "../include/ServerHandler.h"
 
 
-short port = boost::lexical_cast<short>(cmdParams[1].substr(portSeparator, cmdParams[1].size()));
\ No newline at end of file
+ServerHandler::ServerHandler(ConcurrentDataQueues& queues, ClientInventory& inventory) :
+        _connectionHandler(nullptr), _queues(queues), _inventory(inventory) {}
+
+ServerHandler::~ServerHandler() {
+    delete _connectionHandler;
+}
+
+void ServerHandler::split(std::string& str, std::vector <std::string>& subStrs, std::string delimiter) {
+    size_t start = 0U;
+    size_t end = str.find(delimiter);
+    while (end != std::string::npos) {
+        subStrs.push_back(str.substr(start, end - start));
+        start = end + delimiter.length();
+        end = str.find(delimiter, start);
+    }
+    subStrs.push_back(str.substr(start, str.size() - start + 1));
+}
+
+Frame ServerHandler::receiveFrameFromClient() {
+    boost::unique_lock <boost::mutex> lock(_queues.mutexToServer);
+    Frame frame;
+    if (!_queues.framesToServer.empty()) {
+        frame = _queues.framesToServer.front();
+        _queues.framesToServer.pop();
+    }
+    return frame;
+}
```

Comment:
Assuming framesToServer is empty, you would return a new Frame created with the default constructor, also maybe frame should be a reference because copying the frame around seems a little wasteful.

Reply:
a. you're correct, thats why I added the UNINITIALIZED command
b. Frame is pretty chip, I rather not use pointers which are tricky. ref is an option